pkg/resourcemanager/vm/reconcile.go

```diff
@@ -52,11 +54,19 @@ func (g *AzureVirtualMachineClient) Ensure(ctx context.Context, obj runtime.Obje
 	// to overcome the issue with the lack of idempotence of the Create call
 	item, err := g.GetVirtualMachine(ctx, resourceGroup, resourceName)
 	if err == nil {
-		instance.Status.Provisioned = true
-		instance.Status.Provisioning = false
-		instance.Status.Message = resourcemanager.SuccessMsg
-		instance.Status.ResourceId = *item.ID
-		return true, nil
+
+		if *item.ProvisioningState == SucceededProvisioningState {
+
+			instance.Status.Provisioned = true
+			instance.Status.Provisioning = false
+			instance.Status.Message = resourcemanager.SuccessMsg
+			instance.Status.ResourceId = *item.ID
+			return true, nil
+		}
+
```

Comment:
```suggestion
instance.Status.State = *item.ProvisioningState
```

Then in the status.Message you could be more verbose. "The requested resource has been requested but is not yet ready." If you want.

Reply:
added, thanks for the suggestion